### ISSUE TITLE
remove emacs/LICENSE

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5009]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5009|
|Type|task|
|Created|2011-03-29 18:39:22Z|
|Branch|https://github.com/twisted/twisted/tree/emacs-license-5009-2|

the [emacs/LICENSE](emacs/LICENSE) file contains a public domain license, which should be removed since all contributions are (now supposed to be) MIT licensed.

<details><summary>Searchable metadata</summary>

```
trac-id__5009 5009
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_emacs_license_5009_2 branches/emacs-license-5009-2
branch_author__thijs__exarkun thijs, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1301423962000000 1301423962000000
changetime__1306081982000000 1306081982000000
version__None None
owner__thijs thijs
cc__thijs cc__mithrandi
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @washort|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [31486]) Branching to 'emacs-license-5009'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

```
(08:33:15 PM) kenaan: thijs r31486 A(branches/emacs-license-5009/): Branching to 'emacs-license-5009' ...
(08:33:51 PM) hobo_: exarkun: thats my actual question put formally
(08:34:04 PM) exarkun: thijstriemstra: You can't just relicense code that someone else holds the copyright to.
(08:34:22 PM) Thijs Triemstra: thats why it will be assigned to washort?
(08:34:33 PM) Thijs Triemstra: if he doesnt merge than so be it..?
(08:35:05 PM) exarkun: I think it's questionable even to commit such a change.
(08:35:31 PM) deepy: it can spark very heated debate
(08:35:47 PM) Thijs Triemstra: oops. but i thought the coding standard was law ;)
(08:36:05 PM) deepy: commit first, ask later?
(08:36:51 PM) glyph: it's public domain, so, allen can't do anything about it
(08:36:54 PM) Thijs Triemstra: hes also listed first in LICENSE
```

I'm going to commit the changes anyway, and leave it up to washort to commit/ok/reject it by assigning it to him.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [31490]) add license header to all files, remove public domain reference and maintainer email. refs [#5009](https://github.com/twisted/twisted/issues/5009)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [31491]) remove license file (Allen Short is also included in LICENSE), refs [#5009](https://github.com/twisted/twisted/issues/5009)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

I made the changes and added a license header to each file. Assigning to washort for review.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_8"></a>|@mithrandi commented|
|-|-|

FWIW, there's no such thing as a "public domain license". If a work is in the public domain, then there is no longer a copyright holder, thus granting a license is a meaningless act, since you can already distribute it however you want. There is also no requirement to provide any notice that something you are distributing is in the public domain.

In the free software world, people tend to talk about licenses in terms of what restrictions a license implies, not in terms of what the license allows. However, in terms of copyright law, a certain (very broad) set of rights are reserved exclusively for the copyright holder; a copyright license then serves the purpose of a conditional grant of these exclusive rights to others. If the work is in the public domain, then there is nothing wrong or legally questionable providing a license that claims to grant these rights; it's just somewhat meaningless, since everyone already has those rights anyway, even if your so-called license doesn't grant them.

If the work you are distributing is instead a derivative work of the public domain work, then the portion in which someone else's copyright subsists is still relevant, and the license is definitely meaningful as far as that goes.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_9"></a>|@mithrandi set owner to @thijstriemstra|
|-|-|

I guess my previous comment really should have been a review. As I already mentioned, the LICENSE change seems fine, as does everything else; however, this branch needs a news file before it can be merged.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

(In [31568]) add news file, refs [#5009](https://github.com/twisted/twisted/issues/5009)

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_11"></a>|@jonathanj removed owner|
|-|-|

This ticket looks like it should be up for review.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_12"></a>|@mithrandi set owner to @thijstriemstra|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

Fail!  The branch deletes the _top level_ LICENSE file and leaves the emacs/LICENSE file.  The branch doesn't even merge cleanly, since svn cannot merge the LICENSE deletion with the changes to the LICENSE file in trunk since the branch was made.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

(In [31921]) Branching to 'emacs-license-5009-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun set status to `closed`|
|-|-|

(In [31924]) Merge emacs-license-5009-2

Author: thijs, exarkun
Reviewer: mithrandi, exarkun
Fixes: [#5009](https://github.com/twisted/twisted/issues/5009)

Put the emacs/ directory under the same license as the rest of the codebase.
